Gems/EMotionFX/Code/EMotionFX/Rendering/Common/RotateManipulator.cpp

```diff
@@ -405,45 +409,55 @@ namespace MCommon
 
             // the intersection variables
             AZ::Vector3 intersectA, intersectB;
+            AZ::Vector3 rayReciprocal = mousePosRay.GetDirection().GetReciprocal();
+            AZ::Vector3 normalizedDirection;
+            auto intersectRayDirectionAxis =
+                [=](AZ::Aabb& aabb, const MCore::Ray& ray, const AZ::Vector3& dirRCP, AZ::Vector3& normalizedDirection)
```

Comment:
update: Please don't capture with `=`, it makes the code harder to read. This is a little confusing as we have the out parameter called `normalizedDirection`, and the local variable that's reused below (but could actually be captured by the `=` capture clause potentially).

It would instead be better to have this function not capture anything, and just take the parameters as inputs and return an optional, `nullopt` if there's no intersection, and a filled optional containing the normalized direction if it succeeds

Reply:
(for this to compile you'll likely need to specify the return value of the function with `-> AZStd::optional<AZ::Vector3>`

Reply:
kind of difficult since its used if that if else chain below. maybe the conditions can be split into separate functions maybe? It would avoid a lot of these floating variables. 

```
checkXAxis()
checkYAxis()
checkZAxis()
```

Reply:
+1 yeah a small refactor to break stuff up a bit more might be worth it - it's the two variables with the name `normalizedDirection` which have slightly different scope which are a little confusing and could be error-prone